scheds/rust/scx_layered/src/bpf/util.bpf.c

```diff
@@ -54,15 +54,21 @@ static char *format_cgrp_path(struct cgroup *cgrp)
 	return path;
 }
 
-static inline bool match_prefix(const char *prefix, const char *str, u32 max_len)
+int match_prefix(const char *prefix, const char *str, u32 max_len)
 {
-	int c;
+	int c = 0;
+	char *p, *s;
+
+	if (!prefix || !str)
+		return -EINVAL;
 
 	bpf_for(c, 0, max_len) {
-		if (prefix[c] == '\0')
-			return true;
-		if (str[c] != prefix[c])
-			return false;
+		p = MEMBER_VPTR(prefix, [c]);
```

Comment:
This doesn't work correctly, as with this implementation it is returning null so layered matching doesn't work properly.

Reply:
`MEMBER_VPTR()` needs to know the static size of the data structure it's accessing. Here, the compiler doesn't know anything about the size of the areas, so `MEMBER_VPTR()` won't work. I don't know what the right solution here is. Maybe some sort of probe_read? We can ask the BPF folks.